### DISCUSSION TITLE
3.5分包问题

### DISCUSSION THREAD

**Discussion by peakandyuri** [umijs/umi#7611] [Q&A] [answered]
## What happens?
按照官网的快速入门教程创建的项目无法使用分包加载

`mkdir myapp && cd myapp` 创建项目目录
`yarn create @umijs/umi-app` 创建项目
`yarn` 安装

## Mini Showcase Repository(REQUIRED)
使用脚手架直接创建项目的仓库地址
https://gitee.com/peakandyuri/umi3.5chunks.git

## How To Reproduce

修改.umirc.ts文件，在配置后面追加官网分包示例代码
```typescript
export default defineConfig({
     //...省略默认options
     chunks: ['vendors', 'umi'],
     chainWebpack: function (config, { webpack }) {
        //官网是config.merge，效果是一致的
        config.optimization.splitChunks({
	    chunks: 'all',
		minSize: 30000,
		minChunks: 3,//注释掉该行可以分包build成功，但是访问页面是白屏，不会加载chunk的js
		automaticNameDelimiter: '.',
		cacheGroups: {
		    vendor: {
			name: 'vendors',
			test({ resource }) {
				return /[\\/]node_modules[\\/]/.test(resource);
			},
			priority: 10,
		},
	    }
        })
    }
})
```
执行`yarn build`，出现第一个报错，chunks of vendors not found
![image](https://user-images.githubusercontent.com/26075324/143013352-d0a49302-aee8-4413-84c8-361302190008.png)

注释掉`minChunks`的配置，再build可以成功，上传至http服务器
![image](https://user-images.githubusercontent.com/26075324/143013732-74ea326e-322e-42b2-9ff3-5232f23f09d3.png)
访问页面，出现白屏，似乎是只加载了umi.js和vendors.js的脚本，没有加载chunk的原因
![image](https://user-images.githubusercontent.com/26075324/143014582-1d744cf7-c5ef-47fc-b4ac-7192a63763ba.png)

执行`yarn start`启动访问，发现也是白屏，原因应该和build是一样的
![image](https://user-images.githubusercontent.com/26075324/143014227-5a8f0dc0-f69e-4e5f-86f6-352ca1bcb91d.png)

发现issues里也有一些人遇到相同的问题，但是都被关闭了，也没有在里面看到这个问题的解决方案，其中有个人说回退到3.4.18版本解决了白屏的问题，但是我试了依然是白屏的。

## Context

- **Umi Version**: v3.5.20
- **Node Version**: v14.18.0
- **Platform**: windows 11

**Comment by renkunx** [comment, marked as answer]
cacheGroups: {
		    vendors: {
			name: 'vendors',
			test({ resource }) {
				return /[\\/]node_modules[\\/]/.test(resource);
			},
			priority: 10,
		},

**Reply by peakandyuri** [reply]
使用config.merge可以了

**Comment by peakandyuri** [comment]
算了，使用了antd组件，页面就又空白了，还是自己学着搭建webpack项目吧，顺便多了解下底层的东西，用别人的方便是方便，遇到问题不知道怎么解决
